dev/plugins/hu.elte.txtuml.export.uml2/src/hu/elte/txtuml/export/uml2/structural/PackageExporter.xtend

```diff
@@ -26,13 +27,63 @@ abstract class AbstractPackageExporter<S, T extends Package> extends Exporter<S,
 		super(parent);
 	}
 
-	def exportPackageFragment(IPackageFragment packageFragment) {
-		packageFragment.children.forEach[exportCompUnit(it as ICompilationUnit)]
-		val packageRoot = packageFragment.parent as IPackageFragmentRoot
-		val subPackages = packageRoot.children.map[it as IPackageFragment].filter [
-			elementName.startsWith(packageFragment.elementName + ".")
+	def exportPackageFragment(List<IPackageFragment> packageFragments) {
+		if(packageFragments.empty) {
+			return
+		}
+			
+		packageFragments.forEach[packageFragment | 
+			packageFragment.children.forEach[exportCompUnit(it as ICompilationUnit)]
 		]
-		subPackages.forEach[exportPackage[result.nestedPackages += it]]
+		
+		if(packageFragments.size == 1) {
+			val packageFregement = packageFragments.get(0)
+				val subPackages = getSubPackages(packageFregement)
+				subPackages.forEach[p | exportPackage(#[p])[result.nestedPackages += it]]
+				
+		} else if(packageFragments.size == 2) {
+			val subPackages1 = getSubPackages(packageFragments.get(0))
+			val subPackages2 = getSubPackages(packageFragments.get(1))	
+							
+			var i = 0
+			var j = 0			
+			while(i < subPackages1.size || j < subPackages2.size) {
+				
+				if(j >= subPackages2.size || 
+					(i < subPackages1.size && subPackages1.get(i).elementName < subPackages2.get(j).elementName)
```

Comment:
Is it guaranteed that these collections are ordered by `elementName`? I suggest an explicit sorting beforehand.